packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -352,9 +425,32 @@ export class OrderedClientElection
         if (this.isEligibleFn(client)) {
             this._eligibleCount--;
             if (this._electedClient === client) {
-                // Automatically shift to next oldest client
-                const nextClient = this.findFirstEligibleClient(this._electedClient.youngerClient);
-                this.tryElectingClient(nextClient, sequenceNumber);
+                // Removing the _electedClient. There are 2 possible cases:
+                if (this._electedParent !== client) {
+                    // 1. The _electedClient is a summarizer that we've been allowing to finish its work.
+                    // Let the _electedParent become the _electedClient so that it can start its own summarizer.
```

Comment:
It might be helpful to add which step from the scenario in the big comment is this (steps v and vi). Similarly, other points in code correspond to the scenario.
However, this does have the disadvantage that these comments can become outdated if anything changes!